ui/src/state.ts

```diff
@@ -18,6 +26,12 @@ export interface State {
   papers: Readonly<Papers>;
   setPapers(papers: Papers): void;
 
+  /*
+   * USER DATA
+   */
+  userId?: number,
+  userLibrary: UserLibrary | null;
```

Comment:
I think you could do ?UserLibrary which would accept UserLibrary | null | undefined. Not sure if you're strict w/ undefined though

Reply:
This setup doesn't allow for ? before any types, it complains saying "JsDoc types can only be used inside documentation comments". I followed the pattern of other type declarations that specify | null to go around it